internal/provider/repository_branch_protection_resource.go

```diff
@@ -0,0 +1,816 @@
+				Default:     booldefault.StaticBool(false),
+			},
+			"merge_whitelist_usernames": schema.ListAttribute{
+				Description: "Usernames allowed to merge.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"merge_whitelist_teams": schema.ListAttribute{
+				Description: "Teams allowed to merge.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"enable_approvals_whitelist": schema.BoolAttribute{
+				Description: "Enable approvals whitelist.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"approvals_whitelist_usernames": schema.ListAttribute{
+				Description: "Usernames that can approve.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"approvals_whitelist_teams": schema.ListAttribute{
+				Description: "Teams that can approve.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"required_approvals": schema.Int64Attribute{
+				Description: "Number of required approvals.",
+				Optional:    true,
+			},
+			"block_on_rejected_reviews": schema.BoolAttribute{
+				Description: "Block merge on rejected reviews.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"block_on_official_review_requests": schema.BoolAttribute{
+				Description: "Block merge on official review requests.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"block_on_outdated_branch": schema.BoolAttribute{
+				Description: "Block merge on outdated branch.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"dismiss_stale_approvals": schema.BoolAttribute{
+				Description: "Dismiss stale approvals.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+		},
+	}
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *repositoryBranchProtectionResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*forgejo.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf(
+				"Expected *forgejo.Client, got: %T. Please report this issue to the provider developers.",
+				req.ProviderData,
+			),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *repositoryBranchProtectionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	defer un(trace(ctx, "Create repository branch protection resource"))
+
+	var data repositoryBranchProtectionResourceModel
+
+	diags := req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "Create repository branch protection", map[string]any{
```

Comment:
You log the user input, but not the actual API request parameters that are sent. Please log all fields that are written in the `modelToCreateOption()/modelToEditOption()` functions.

Please check https://github.com/svalabs/terraform-provider-forgejo/blob/edff37edbd29028fad2f3a4a2b00a7da5fcfc37a/internal/provider/user_resource.go#L346 for an example.